File: src/org/usfirst/frc5293/commands/teleop/control/ShooterControl.java
package org.usfirst.frc5293.commands.teleop.control;

import org.usfirst.frc5293.Input;
import org.usfirst.frc5293.Subsystems;
import org.usfirst.frc5293.commands.util.ContinuousCommand;
import org.usfirst.frc5293.commands.util.LimitFunction;

public class ShooterControl extends ContinuousCommand {

    public ShooterControl() {
        requires(Subsystems.getShooter());
    }

    @Override
    protected void execute() {
        final double deadzone = 0.10;
        final double raw = Input.getShooter().getJoystick().getY();

        if (Math.abs(raw) > deadzone) {
            Subsystems.getShooter().setPower(raw);
        } else {
            Subsystems.getShooter().stop();
        }
    }

    @Override
    protected void end() {
        Subsystems.getShooter().stop();
    }
}

File: src/org/usfirst/frc5293/commands/teleop/control/ShooterControl.kt
package org.usfirst.frc5293.commands.teleop.control

import org.usfirst.frc5293.Input
import org.usfirst.frc5293.Subsystems
import org.usfirst.frc5293.commands.util.EmptyCommand

class ShooterControl : EmptyCommand() {

    init {
        requires(Subsystems.shooter)
    }

    override fun execute() {
        val deadzone = 0.10
        val raw = Input.shooter.joystick.y

        if (Math.abs(raw) > deadzone) {
            Subsystems.shooter.power = raw
        } else {
            Subsystems.shooter.stop()
        }
    }

    override fun end() {
        Subsystems.shooter.stop()
    }
}
